vllm/model_executor/models/llama.py

```diff
@@ -487,6 +489,65 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
                 except KeyError:
                     pass
 
+    def load_quantized_weights(self, model_name_or_path: str,
+                               quant_config: QuantizationConfig):
+        weights = safetensors_weights_iterator([
+            os.path.join(model_name_or_path,
+                         quant_config.quantized_weights_path)
+        ])
+        params_dict = dict(self.named_parameters())
+        quant_shards = [
+            (param_name, weight_name,
+             quant_config.shard_layers[param_name][weight_name])
+            for param_name in quant_config.shard_layers
+            for weight_name in quant_config.shard_layers[param_name]
+        ]
+        quant_map = [(param_name, quant_config.quant_layers[param_name])
```

Comment:
If the challenge here is supporting different integrations in the same place (e.g. Quark + AMMO), I would prefer to ahve `quark.py` and `ammo.py` rather than pushing this logic into weight loading. Since then we separate the feature from the model definition